server/services/v1/cache/query_runner.go

```diff
@@ -0,0 +1,312 @@
+}
+
+func (runner *SetQueryRunner) Run(ctx context.Context, tenant *metadata.Tenant) (*CacheResponse, error) {
+	project, err := tenant.GetDatabase(ctx, metadata.NewDatabaseNameWithBranch(runner.req.GetProject(), metadata.MainBranch))
+	if err != nil {
+		return nil, err
+	}
+	encodedCacheTableName, err := runner.encoder.EncodeCacheTableName(tenant.GetNamespace().Id(), project.Id(), runner.req.GetName())
+	if err != nil {
+		return nil, err
+	}
+
+	options := &cache.SetOptions{
+		NX: runner.req.GetNx(),
+		XX: runner.req.GetXx(),
+		EX: time.Duration(runner.req.GetEx()) * time.Second,
+	}
+	err = runner.cacheStore.Set(ctx, encodedCacheTableName, runner.req.GetKey(), internal.NewCacheData(runner.req.GetValue()), options)
+
+	if err != nil {
+		return nil, errors.Internal("Failed to invoke set, reason %s", err.Error())
+	}
+	return &CacheResponse{
+		Status: database.InsertedStatus,
+	}, nil
+}
+
+func (runner *GetQueryRunner) Run(ctx context.Context, tenant *metadata.Tenant) (*CacheResponse, error) {
+	project, err := tenant.GetDatabase(ctx, metadata.NewDatabaseNameWithBranch(runner.req.GetProject(), metadata.MainBranch))
+	if err != nil {
+		return nil, err
+	}
+	encodedCacheTableName, err := runner.encoder.EncodeCacheTableName(tenant.GetNamespace().Id(), project.Id(), runner.req.GetName())
+	if err != nil {
+		return nil, err
+	}
+
+	options := &cache.GetOptions{
+		Expiry:    0,
+		GetDelete: false,
+	}
+	data, err := runner.cacheStore.Get(ctx, encodedCacheTableName, runner.req.GetKey(), options)
+	if err != nil {
+		if err == cache.ErrKeyNotFound {
+			return nil, errors.NotFound(err.Error())
+		}
+		return nil, err
+	}
+	return &CacheResponse{
+		Data: data.GetRawData(),
+	}, nil
+}
+
+func (runner *DelQueryRunner) Run(ctx context.Context, tenant *metadata.Tenant) (*CacheResponse, error) {
+	project, err := tenant.GetDatabase(ctx, metadata.NewDatabaseNameWithBranch(runner.req.GetProject(), metadata.MainBranch))
+	if err != nil {
+		return nil, err
+	}
+	encodedCacheTableName, err := runner.encoder.EncodeCacheTableName(tenant.GetNamespace().Id(), project.Id(), runner.req.GetName())
+	if err != nil {
+		return nil, err
+	}
+
+	deletedCount, err := runner.cacheStore.Delete(ctx, encodedCacheTableName, runner.req.GetKey())
+	if err != nil {
+		return nil, errors.Internal("Failed to invoke del, reason %s", err.Error())
+	}
+	return &CacheResponse{
+		Status:       database.DeletedStatus,
+		DeletedCount: deletedCount,
+	}, nil
+}
+
+func (runner *KeysQueryRunner) Run(ctx context.Context, tenant *metadata.Tenant) (*CacheResponse, error) {
+	project, err := tenant.GetDatabase(ctx, metadata.NewDatabaseNameWithBranch(runner.req.GetProject(), metadata.MainBranch))
+	if err != nil {
+		return nil, err
+	}
+	encodedCacheTableName, err := runner.encoder.EncodeCacheTableName(tenant.GetNamespace().Id(), project.Id(), runner.req.GetName())
+	if err != nil {
+		return nil, err
+	}
+
+	pattern := runner.req.GetPattern()
+	if pattern == "" {
+		pattern = "*"
```

Comment:
Here as well there will be lot of keys and it would be expensive, we also need to think about supporting pagination here.

Reply:
yes we will iterate on it.